src/particle/corsika/gen_conversion_table.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) 2018-2022, Eduardo Rodrigues and Henry Schreiner.
+#
+# Distributed under the 3-clause BSD license, see accompanying file LICENSE
+# or https://github.com/scikit-hep/particle for details.
+"""
+Function to generate .csv conversion table from corsica to pdg_id and vice-versa.
+This script should be kept, so the table won't need to be hand-edited in the future.
+"""
+from __future__ import annotations
+
+import csv
+import pathlib
+
+from ..particle import Particle, ParticleNotFound
+from ..shared_literals import common_particles
+
+corsica_pdg_id = [
+    (1, common_particles["gamma"]),
+    (50, common_particles["omega_782"]),
+    (2, common_particles["e_plus"]),
+    (51, common_particles["rho_770_0"]),
+    (3, common_particles["e_minus"]),
+    (52, common_particles["rho_770_plus"]),
+    (53, common_particles["rho_770_minus"]),
+    (5, common_particles["mu_plus"]),
+    (54, common_particles["Delta_1232_pp"]),
+    (6, common_particles["mu_minus"]),
+    (55, common_particles["Delta_1232_plus"]),
+    (7, common_particles["pi_0"]),
+    (56, common_particles["Delta_1232_0"]),
+    (8, common_particles["pi_plus"]),
+    (57, common_particles["Delta_1232_minus"]),
+    (9, common_particles["pi_minus"]),
+    (58, common_particles["Delta_1232_mm_bar"]),
+    (10, common_particles["K_L_0"]),
+    (59, common_particles["Delta_1232_minus_bar"]),
+    (11, common_particles["K_plus"]),
+    (60, common_particles["Delta_1232_0_bar"]),
+    (12, common_particles["K_minus"]),
+    (61, common_particles["Delta_1232_plus_bar"]),
+    (13, common_particles["neutron"]),
+    (62, common_particles["Kst_892_0"]),
+    (14, common_particles["proton"]),
+    (63, common_particles["Kst_892_plus"]),
+    (15, common_particles["antiproton"]),
+    (64, common_particles["Kst_892_minus"]),
+    (16, common_particles["K_S_0"]),
+    (65, common_particles["Kst_892_0_bar"]),
+    (17, common_particles["eta"]),
+    (66, common_particles["nu_e"]),
+    (18, common_particles["Lambda"]),
+    (67, common_particles["nu_e_bar"]),
+    (19, common_particles["Sigma_plus"]),
+    (68, common_particles["nu_mu"]),
+    (20, common_particles["Sigma_0"]),
+    (69, common_particles["nu_mu_bar"]),
+    (21, common_particles["Sigma_minus"]),
+    (22, common_particles["Xi_0"]),
+    (71, "η → γγ"),
```

Comment:
So in these cases the correspondence is not a PDGID but rathera decay name? Guess this is necessary specifically for Corsika7. Maybe worth a common above in the definition of this long list.

Reply:
yes, you are right. I am going to implement all your longer suggestions, when I find time :)

Reply:
OK, sounds good. There is no rush but asap is perfect :-).